src/reducers/cds-services-reducers.js

```diff
@@ -0,0 +1,62 @@
+import isEqual from 'lodash/isEqual';
+import * as types from '../actions/action-types';
+
+const initialState = {
+  configuredServices: {},
+  defaultUrl: 'https://fhir-org-cds-services.appspot.com/cds-services',
+  // defaultUrl: 'http://localhost:3000/cds-services',
+  testServicesUrl: null,
+};
+
+const cdsServicesReducers = (state = initialState, action) => {
+  if (action.type) {
+    switch (action.type) {
+      // Signal that the application is retrieving CDS Services
+      case types.DISCOVER_CDS_SERVICES: {
+        if (action.testUrl) {
+          return Object.assign({}, state, { testServicesUrl: action.testUrl });
+        }
+        break;
+      }
+      // Store CDS Services grabbed successfully from a discovery endpoint
+      case types.DISCOVER_CDS_SERVICES_SUCCESS: {
+        if (action.services) {
+          const incomingServices = {};
+
+          // For each service returned by the discovery endpoint, check if it is already configured (or not changed),
+          // and if not, store the service definitions by the specific CDS Service URL in the
+          // configuredServices property of the reducer state
+          action.services.forEach((service) => {
+            const serviceUrl = `${state.testServicesUrl}/${service.id}`;
+            const serviceEndpoint = state.configuredServices[serviceUrl];
+            if (!serviceEndpoint || !isEqual(serviceEndpoint, service)) {
+              const serviceCopy = Object.assign({}, service);
+              serviceCopy.enabled = true;
+              incomingServices[serviceUrl] = serviceCopy;
+            }
+          });
+
+          const newServicesKeys = Object.keys(incomingServices);
+
+          if (newServicesKeys.length) {
+            const newServices = Object.assign({}, state.configuredServices);
```

Comment:
Since this is a shallow copy just ensure that its not updating the existing contents of configuredServices because the inner contents of the ```configuredServices``` hash would be by reference.

Reply:
We want to update all deep contents of a key in this hash if the existing service exists (situation for replacing a stale configured CDS Service with a potentially updated one), so it shouldn't be an issue here.